src/impls/chat.rs

```diff
@@ -267,6 +267,62 @@ impl ChatServer {
 impl chat_service_server::ChatService for ChatServer {
     type Error = ServerError;
 
+    #[rate(1, 5)]
+    async fn update_all_channel_order(
+        &self,
+        request: Request<UpdateAllChannelOrderRequest>
+    ) -> Result<(), Self::Error> {
+        auth!();
+
+        let UpdateAllChannelOrderRequest {
+            guild_id,
+            channel_ids,
+        } = request.into_parts().0;
+
+        self.chat_tree.check_guild_user(guild_id, user_id)?;
+        self.chat_tree
+            .check_perms(guild_id, 0, user_id, "channels.manage.move", false)?;
+
+        for channel_id in &channel_ids {
+            let id = *channel_id;
+
+            let exists = self.chat_tree.does_channel_exist(guild_id, id);
+
+            if !exists {
+                return Err(ServerError::NoSuchChannel{guild_id, channel_id: id})
+            }
+        }
+
+        let prefix = make_guild_chan_prefix(guild_id);
+        let channels = self
+            .chat_tree
+            .chat_tree
+            .scan_prefix(&prefix)
+            .flat_map(|res| {
+                let (key, value) = res.unwrap();
+                (key.len() == prefix.len() + size_of::<u64>())
+                    .then(|| {
+                        (true)
+                        // Safety: this unwrap is safe since we only store valid Channel message
+                        .then(|| unsafe { Channel::decode(value.as_ref()).unwrap_unchecked() })
```

Comment:
```suggestion
                        // Safety: this unwrap is safe since we only store valid Channel message
                        unsafe { Channel::decode(value.as_ref()).unwrap_unchecked()
```